common/djangoapps/util/enterprise_helpers.py

```diff
@@ -442,3 +446,41 @@ def get_dashboard_consent_notification(request, user, course_enrollments):
             }
         )
     return ''
+
+
+def is_course_in_enterprise_catalog(site, course_id, user, enterprise_catalog_id):
+    """
+    Verify that the provided course id exists in the site base list of course
+    run keys from the provided enterprise course catalog.
+
+    Arguments:
+        course_id (str): The course ID.
+        site: (django.contrib.sites.Site) site instance
+        enterprise_catalog_id (Int): Course catalog id of enterprise
+
+    Returns:
+        Boolean
+
+    """
+    cache_key = get_cache_key(
```

Comment:
It does not seems me useful in our case. we are explicitly calling the `course_discovery_api_client(user=user).catalogs(enterprise_catalog_id).contains.get(course_run_id=course_id)` in our call